internal/core/metadata/application/device.go

```diff
@@ -63,6 +67,16 @@ func AddDevice(d models.Device, ctx context.Context, dic *di.Container) (id stri
 		correlation.FromContext(ctx),
 	)
 
+	// If device is successfully created, check each AutoEvent interval value and display a warning if it's smaller than the suggested 10ms value
+	for _, autoEvent := range d.AutoEvents {
+		valid, err := utils.CheckDuration(autoEvent.Interval, minAutoEventInterval)
+		if err != nil {
+			lc.Warnf("failed to check auto event interval: %v", err)
+		} else if !valid {
+			lc.Warnf("the interval value '%s' is smaller than the suggested value '%s', which might cause abnormal CPU increase", autoEvent.Interval, minAutoEventInterval)
```

Comment:
since the log message is the same, move the warning log to the checking fucntion.